### PR TITLE
Gh1585 allowing \fpeval in array preamble

### PR DESCRIPTION
**Pull request by FrankMittelbach**
# Internal housekeeping

## Status of pull request

 Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [n/a] Rollback provided (if necessary)?
- [x] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

There is one surprising test suite diff for tagging-002-longtable. That needs checking and verification.

### CODE REVIEW

**Comment by FrankMittelbach** [comment]
I've done some minimal adjustment to longtable, to make it work with the change in array. It could probably be simplified by using \@startpbox@preamble and then keeping \@startpbox for the action rather than having \@startpbox and \@startpboxaction with the same top-level definitions. But  that might mean more checking on other packages how they use \@startpbox and \@mkpream. So I'm not sure it is worth the effort. @davidcarlisle does it looks okay to you?

File: required/tools/array.dtx
Comment (Skillmon) [inline required/tools/array.dtx:3308]:
```suggestion
%    preamble built-up \cs{@startpbox@action} does the actual work.
```

File: required/tools/array.dtx
Comment (Skillmon) [inline required/tools/array.dtx:3312]: Something strange happened here (and in the line above), looks like you rewrote that sentence and forgot to delete the old version.

**Review by josephwright — CHANGES_REQUESTED** [review]
We seem to have two different things going on here: the `changes.txt` edit it nothing to do with the rest of the PR!

**Comment by FrankMittelbach** [comment]
> We seem to have two different things going on here: the `changes.txt` edit it nothing to do with the rest of the PR!

when I fixed 1418 I had by mistake this PR checked out, which is why it now contains both the changes for that (documented in base) as well as the change for this PR (documented in changes of tools). I was too lazy to back it out, so this PR contains now the fixes for both issues, hence the unrelated changes text and the fixes to doc.dtx

**Review by josephwright — APPROVED** [review]

**Review by davidcarlisle — APPROVED** [review]

**Review by car222222 — COMMENTED** [review]
I have now, too late?, found some (probably)necessary corrections here.

I cannot now Approve it, not sure why?

File: base/doc/ltnews41.tex
Comment (FrankMittelbach) [reply base/doc/ltnews41.tex:409]: not sure, they have been expanded but aren't any longer, so "get" seems wrong "got expanded" perhaps?

**Comment by FrankMittelbach** [comment]
> I have now, too late?, found some (probably) necessary corrections here.

added (with one query)

> I cannot now Approve it, not sure why?

same reason as before ... after a while it got merged (two approvals), then you can only comment but not review any longer.

**Comment by car222222** [comment]
Understood, but the app is a bit confused about this: it still has the normal "Review" button and functionality, up to the final stage where there are no options for Approval, etc. (And it does not appear as a "Review", although within the change files it looks like one.)

File: base/doc/ltnews41.tex
Comment (car222222) [reply base/doc/ltnews41.tex:409]: Yes, my typo! I meant "got".

**Comment by FrankMittelbach** [comment]
yes I agree it is a bit odd; your only indication is that it says merged on top (and then in the end when you can't actually finish the review) - happened to me a few times as well. But at least the comments get added and distributed so all that is lost is really to be able to say "approved" or "changes requested" which kind of makes sense because the merge has already happened.

File: base/doc/ltnews41.tex
Comment (FrankMittelbach) [reply base/doc/ltnews41.tex:409]: ok updated

**Comment by car222222** [comment]
Agreed that it makes sense!  But it would be better if it were explained in detail somewhere (maybe it is!??).  

Actually this seems to be a general problem with apps and the like these days: it is seldom clear how or where to find simple explanations of how they can, or should, be used.  

And, especially in iOS apps, “buttonless”(or minimal buttons) seems to be the new normal and preferred interface.

**Pull request merged**
